apps/site/assets/ts/hooks/useFetch.ts

```diff
@@ -0,0 +1,25 @@
+import useSWR from "swr";
+import { fetchJsonOrThrow } from "../helpers/fetch-json";
+import { FetchState, FetchStatus } from "../helpers/use-fetch";
+
+/**
+ * Generic helper function for fetching JSON from a URL. Specify the expected
+ * return type as the type parameter, e.g
+ * `useFetch<Schedule[]>("/api/schedules/or/something")`
+ * @param {string} backendUrl - The path, usually starting with /api/, that our
+ * backend will respond with data from.
+ * @returns {{ status: FetchStatus, data: T }} An object containing the fetch
+ * status, and the returned data.
+ */
+const useFetch = <T>(backendPath: string): FetchState<T> => {
+  const { data, error } = useSWR<T>(
+    backendPath,
+    async (url: string): Promise<T> => fetchJsonOrThrow(url)
+  );
+  if (error) {
+    return { status: FetchStatus.Error };
+  }
+  return { status: FetchStatus.Data, data };
+};
```

Comment:
I like the idea of doing this.  I feel like this pull request is the wrong place for it.
It would make sense to do this update as its own pull request, and update the usage across all of our web requests.  At least for the stop page redesign.  We could also discuss things about error state handling.  Something like this would allow a global way handle errors (show a toast or banner) or loading (display the dots) so the components don't need to worry about it.

Reply:
I'm happy to follow up with a separate PR migrating all the fetches in the stop page redesign to this, am less happy about removing this and having to scaffold files and tests for a useOneWayFares hook... 😂

Reply:
We should add a follow up ticket to try and have this handle the error and loading states so each page/component doesn't have to handle it separately and help towards a more consistent user experience

Reply:
@kotva006 About to write ticket, have WIP migration over in #1667 - lots to think about!

Reply:
I ended up separating this out into this PR, which also migrates the stop page redesign into using it - #1667